cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -652,8 +653,15 @@ private V1ObjectMeta createJob(V1ObjectMeta metadata, Map<String, RuntimeSpecifi
         .endTemplate()
       .endSpec()
       .build();
-
-    batchV1Api.createNamespacedJob(kubeNamespace, job, "true", null, null);
+    try {
+      batchV1Api.createNamespacedJob(kubeNamespace, job, "true", null, null);
+    } catch (ApiException e) {
+      if (e.getCode() == ERROR_CODE_ALREADY_EXISTS) {
+        LOG.warn("The job already exists : " , e.getResponseBody());
```

Comment:
`The kubernetes job {} already exists. Ignoring  resubmission of the job. `

Reply:
done :)